packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTree.tsx

```diff
@@ -40,6 +44,7 @@ interface ModelsTreeOwnProps {
   hierarchyConfig?: Partial<ModelsTreeHierarchyConfiguration>;
   visibilityHandlerOverrides?: ModelsTreeVisibilityHandlerOverrides;
   filter?: string;
+  getFilteredPaths?: (props: { imodelAccess: IModelAccess; idsCache: ModelsTreeIdsCache; hierarchyConfig: ModelsTreeHierarchyConfiguration }) => FilteredPaths;
```

Comment:
I wonder if we really need to provide these props. What do you think about instead providing the following?

```ts
type HierarchyFilteringPaths = Awaited<ReturnType<Required<VisibilityTreeProps>["getFilteredPaths"]>>;

getFilteredPaths?: (props: {
    createInstanceKeyPaths: (props: { keys: Array<InstanceKey | ElementsGroupInfo>; } | { label: string; }) => Promise<HierarchyFilteringPaths>
}) => Promise<HierarchyFilteringPaths>;
```

This would even allow us to avoid exposing `ModelsTreeIdsCache` and some other currently internal apis.

Reply:
Passed `createInstanceKeyPaths` to `getFilteredPaths` to avoid the need to provide these props.